Src/Newtonsoft.Json.UnityConverters.Tests/ConvertingUnityTypes/TypeTester.cs

```diff
@@ -90,16 +141,71 @@ public void SerializesArrayAsExpected((T input, object anonymous) representation
         public void DeserializesArrayAsExpected((T expected, object anonymous) representation)
         {
             // Arrange
-            JsonSerializerSettings settings = GetSettings();
-            string input = JArray.FromObject(new [] { representation.anonymous }).ToString(Formatting.None);
+            string input = Serialize(new[] { representation.anonymous });
 
             // Act
-            T[] result = JsonConvert.DeserializeObject<T[]>(input, settings);
+            T[]? result = Deserialize<T[]>(input);
 
             // Assert
             Assert.IsNotNull(result);
-            Assert.AreEqual(1, result.Length, "result.Length");
-            AssertAreEqual(representation.expected, result[0]);
+            Assert.AreEqual(1, result!.Length, "result.Length");
+            AssertAreEqual(representation.expected, result[0], $"Input given: '{input}'");
+        }
+
+        [Test]
+        public virtual void OkWithEmptyObject()
+        {
+            // Arrange
+            string input = "{}";
+
+            // Act
+            Assert.DoesNotThrow(() =>
+            {
+                _ = Deserialize(input);
+            });
+        }
+    }
+
+    public abstract class ValueTypeTester<T> : TypeTester<T>
+        where T : struct
+    {
+        [Test]
+        [TestCaseSource("representations")]
+        public void SerializesNullableAsExpected((T input, object anonymous) representation)
+        {
+            // Arrange
+            string expected = Serialize(representation.anonymous);
+
+            // Act
+            string result = Serialize((T?)representation.input);
+
+            // Assert
+            Assert.AreEqual(expected, result);
+        }
+
+        [Test]
+        [TestCaseSource("representations")]
+        public void DeserializesNullableValueAsExpected((T expected, object anonymous) representation)
+        {
+            // Arrange
+            string input = Serialize(representation.anonymous);
+
+            // Act
+            T? result = Deserialize<T?>(input);
+
+            // Assert
+            Assert.IsNotNull(result);
+            AssertAreEqual(representation.expected, result!.Value);
+        }
+
+        [Test]
+        public void DeserializesNullableNullAsExpected()
+        {
+            // Act
+            T? result = Deserialize<T?>("null");
+
+            // Assert
+            Assert.IsNull(result);
         }
```

Comment:
Perhaps a "OkWithSerializingNullableNull" as well? Think that Json.NET actually tosses the null value into the JsonConverters still. Needs doublechecking